ConnectorRevit/ConnectorRevit/UI/ConnectorBindingsRevit.Receive.cs

```diff
@@ -132,41 +127,32 @@ await elementTypeMapper
 
           previousObjects.AddConvertedElements(convertedObjects);
           transactionManager.Finish();
-          return (true, null);
         }
-        catch (Exception ex)
+        catch (OperationCanceledException) when (progress.CancellationToken.IsCancellationRequested)
+        {
+          transactionManager.RollbackAll();
+          throw;
+        }
+        catch (SpeckleNonUserFacingException ex)
         {
           SpeckleLog.Logger.Error(ex, "Rolling back connector transaction");
-
-          string message = $"Fatal Error: {ex.Message}";
-          if (ex is OperationCanceledException)
-          {
-            message = "Receive cancelled";
-          }
-
-          progress.Report.LogOperationError(new Exception($"{message} - Changes have been rolled back", ex));
-
           transactionManager.RollbackAll();
-          return (false, ex); //We can't throw exceptions in from RevitTask, but we can return it along with a success status
```

Comment:
@connorivy we were doing this (returning rather than throwing) for a reason. Is this no longer needed since we switched away from `RevitTask`?